### ISSUE TITLE
Fusing task-specific and task-agnostic adapters

### ISSUE THREAD

**Issue by murthyrudra** [adapter-hub/adapter-transformers#469]
## Environment info
- `adapter-transformers` version: 3.1.0
- Platform: Linux-4.18.0-425.3.1.el8.x86_64-x86_64-with-glibc2.17
- Python version: 3.8.11
- Huggingface_hub version: 0.11.1
- PyTorch version (GPU?): 1.12.1 (False)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: yes
- Using distributed or parallel set-up in script?: no
     

## Details

Hi, I am trying to combine `task-specific` and `task-agnostic` adapters. Assume I have three tasks `Task-A`, `Task-B`, and, `Task-C`.  I will add task-specific adapters and task-agnostic adapters as follows
```
import transformers.adapters.composition as ac

model.add_adapter("TASK-A")
model.add_adapter("TASK-B")
model.add_adapter("TASK-C")

model.add_adapter("TASK-Agnostic")
```
Now I want to fuse the task-specific adapter and task-agnostic adapter dynamically i.e, depending on what the task is. 

Should I fuse the adapters as follows?

```
model.add_adapter_fusion(["TASK-A", "TASK-Agnostic"])
model.add_adapter_fusion(["TASK-B", "TASK-Agnostic"])
model.add_adapter_fusion(["TASK-C", "TASK-Agnostic"])
```

Inside the `forward_pass` of Trainer, I will set the active adapters as follows
```
task_name = get_task_name()
model.active_adapters = ac.Fuse(task_name, "TASK-Agnostic")
```

Is this the right way to implement this? 

Thanks

**Comment by hSterz** [comment]
Hey @murthyrudra, the setup looks good to me. If you want to train any of the adapters or the fusion module you need to make sure that the correct weights are frozen and only the weights you want to train are adapted (you can check out [this](https://github.com/Adapter-Hub/adapter-transformers/blob/master/examples/pytorch/adapterfusion/run_fusion_glue.py#L190) example script).

**Comment by adapter-hub-bert** [comment]
This issue has been automatically marked as stale because it has been without activity for 90 days. This issue will be closed in 14 days unless you comment or remove the stale label.

**Comment by adapter-hub-bert** [comment]
This issue was closed because it was stale for 14 days without any activity.
